### PR TITLE
[FIX] mrp_subcontracting_account: update inventory valuation

### PR DESCRIPTION
**Pull request by Whenrow**
This reverts commit cfd68df.
The fix was to not value the subcontracting location while it has to be
valued.

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by gaikaz** [comment]
Was struggling with this for a second time in a week with a colleague... 🤦🏻
Must also add, that the mentioned commit didn't consider child subcontracting locations, or rather it looked at the company default, but there is an option to set a custom one on each subcontractor contact.
Secondly, the commit actually was onto something. It seems that productions before didn't create account.move records.
Is that right?

File: addons/sale_stock/models/sale_order.py [hunk 1]
```diff
@@ -157,7 +157,7 @@ def action_cancel(self):
             if sale_order.state == 'sale' and sale_order.order_line:
                 sale_order_lines_quantities = {order_line: (order_line.product_uom_qty, 0) for order_line in sale_order.order_line}
                 documents = self.env['stock.picking']._log_activity_get_documents(sale_order_lines_quantities, 'move_ids', 'UP')
-        self.mapped('picking_ids').action_cancel()
+        self.mapped('picking_ids').filtered(lambda p: p.state != 'done').action_cancel()
```
Comment (gaikaz) [inline addons/sale_stock/models/sale_order.py:160]: This change from `sale_stock` sneaked in accidentally?

**Comment by Whenrow** [comment]
@gaikaz Hello,
Any internal location in your company but not under a warehouse can be used as a subcontracting location. At least for the valuation part. The key is to have products into those locations tracked in your company valuation but not in your stock quantities.

The commit that this PR revert was wrong. We retested multiple flows, and it must be like this. I suspect a misconfiguration in a customer database that wrongly inspire us to do the wrong patch. 

The valuation entries will be created when components go from subcontracting location to production location and when the finished product goes from production location to subcontracting location. The moves from your stock to subcontracting (as well as the opposite direction) should not be valued as it's stay in the same company in an accounting perspective.

File: addons/sale_stock/models/sale_order.py (hunk 1 above)
Comment (Whenrow) [reply addons/sale_stock/models/sale_order.py:160]: During testing, I set my warehouse in 2 steps delivery. Marked as done my first transfer. At this time I could not cancel the sales order. We made this fix better in V14+ (see 4c36a35bf14f) but I though this was acceptable to put in v13.0

Comment (gaikaz) [reply addons/sale_stock/models/sale_order.py:160]: I see... I suppose it's fine then. This just looked out of place 😛

**Comment by amoyaux** [comment]
robodoo r+

**Comment by amoyaux** [comment]
robodoo rebase-ff

File: addons/sale_stock/models/sale_order.py (hunk 1 above)
Comment (celm1990) [reply addons/sale_stock/models/sale_order.py:160]: Hi, cancel a picking done is wrong for me, is inconsistent(sale order `canceled `and stock `validated`), please consider keep as V12(at least on stable version)

V12

imagen

**Pull request closed without merging**